packages/kit/src/runtime/client/client.js

```diff
@@ -1548,7 +1548,7 @@ export function create_client({ target }) {
 					redirect_chain: [],
 					details: {
 						state: {},
-						replaceState: url.href === location.href
+						replaceState: options.replace_state || url.href === location.href
```

Comment:
I didn't make the corresponding `form` changes because I haven't used SvelteKit forms yet - so I don't think I'm best placed to make/verify any changes in that area!